pipelines/controller/src/main/java/org/openmrs/analytics/metrics/TextFormat.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020-2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openmrs.analytics.metrics;
+
+import io.prometheus.client.Collector;
+import io.prometheus.client.Collector.MetricFamilySamples;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Enumeration;
+
+public class TextFormat {
+
+  /** Content-type for Data Pipes text version 0.0.1. */
+  public static final String CONTENT_TYPE_001 = "text/plain; version=0.0.1; charset=utf-8";
+
+  /** Write out the text version of the given MetricFamilySamples. */
+  public static void write(Writer writer, Enumeration<MetricFamilySamples> mfs) throws IOException {
+    while (mfs.hasMoreElements()) {
+      Collector.MetricFamilySamples metricFamilySamples = mfs.nextElement();
+      String name = metricFamilySamples.name;
+      writer.write("# TYPE ");
+      writer.write(name);
+      if (metricFamilySamples.type == Collector.Type.COUNTER) {
+        writer.write("_total");
+      }
+      if (metricFamilySamples.type == Collector.Type.INFO) {
+        writer.write("_info");
+      }
+      writer.write(' ');
+      writer.write(typeString(metricFamilySamples.type));
+      writer.write('\n');
+      for (Collector.MetricFamilySamples.Sample sample : metricFamilySamples.samples) {
```

Comment:
This is fine but if it is not too much work, I would have preferred the usual MVC approach that Spring supports well. So in that case we will have a table of key+values that are automatically created from a map in the model using the [`th:each` construct like here](https://www.thymeleaf.org/doc/tutorials/2.1/thymeleafspring.html#listing-seed-starter-data). That will remove a lot of boilerplate code in this class but as I said, only do this if it is not too hard to fit templates here.

Reply:
I tried implementing this, to use [th:each construct like here](https://www.thymeleaf.org/doc/tutorials/2.1/thymeleafspring.html#listing-seed-starter-data), the entire page had to be reloaded to reflect the metrics. For now, I have kept the old implementation, we can discuss more on this.